packages/smooth_app/lib/pages/navigator/app_navigator.dart

```diff
@@ -0,0 +1,332 @@
+            return AppRoutes.EXTERNAL(
+              path[0] == '/' ? path.substring(1) : path,
+            );
+          }
+        }
+
+        return state.location;
+      },
+      errorBuilder: (_, GoRouterState state) => ErrorPage(
+        url: state.location,
+      ),
+    );
+  }
+
+  late GoRouter router;
+
+  // Indicates whether [_initAppLanguage] was already called
+  bool isInitialized = false;
+
+  void _initAppLanguage(BuildContext context) {
+    final UserPreferences userPreferences = context.read<UserPreferences>();
+    ProductQuery.setLanguage(context, userPreferences);
+    context.read<ProductPreferences>().refresh();
+
+    // The migration requires the language to be set in the app!
+    context.read<SmoothAppDataImporter>().startMigrationAsync();
+  }
+
+  /// Extract the barcode from a path only if the route have at least 8 digits
+  /// in the second part (we don't care about extra elements)
+  /// Some examples:
+  /// - produit/156164894948
+  /// - product/3017620422003/nutella-ferrero
+  String? _extractProductBarcode(String path) {
+    if (path.isEmpty) {
+      return null;
+    }
+
+    final List<String> pathParams = path.split('/').sublist(1);
+
+    if (pathParams.length > 1) {
+      final String barcode = pathParams[1];
+
+      // Ensure we only have digits and at least 8 characters
+      if (int.tryParse(barcode) != null && barcode.length >= 8) {
+        return barcode;
+      }
+    }
+
+    return null;
+  }
+
+  bool _isAnInternalRoute(String path) {
+    if (path == HOME_PAGE) {
+      return true;
+    } else {
+      return path.startsWith('/_');
+    }
+  }
+
+  //region Onboarding
+
+  bool _isOnboardingComplete(BuildContext context) {
+    return _getCurrentOnboardingPage(context).isOnboardingComplete();
+  }
+
+  Widget _findLastOnboardingPage(BuildContext context) {
+    return _getCurrentOnboardingPage(context).getPageWidget(context);
+  }
+
+  OnboardingPage _getCurrentOnboardingPage(BuildContext context) {
+    final UserPreferences userPreferences = context.read<UserPreferences>();
+
+    final OnboardingPage lastVisitedOnboardingPage =
+        userPreferences.lastVisitedOnboardingPage;
+    return lastVisitedOnboardingPage;
+  }
+
+  //endregion Onboarding
+
+  /// Internal routes
+  /// To differentiate external routes (eg: /product/12345678), we prefix all
+  /// internal routes with an underscore
+  static const String HOME_PAGE = '/';
+  static const String PRODUCT_DETAILS_PAGE = '_product';
+  static const String PRODUCT_LOADER_PAGE = '_product_loader';
+  static const String PRODUCT_CREATOR_PAGE = '_product_creator';
+  static const String PREFERENCES_PAGE = '_preferences';
+  static const String SEARCH_PAGE = '_search';
+  static const String EXTERNAL_PAGE = '_external';
+}
+
+/// A list of internal routes to use with [AppNavigator]
+// TODO(g123k): Improve this with sealed classes (Dart 3 is required)
+// ignore_for_file: non_constant_identifier_names
```

Comment:
Looks like in `dart` the constants should rather be written in camelCase. Is that the same with sealed classes?

Reply:
Sealed classes are actually like regular classes.
This is just a temporary way to distinguish a simple getter from this.
I know this isn't perfect